From 841c3421db481b69aca9de3635c7caff3ee14a7b Mon Sep 17 00:00:00 2001
From: Tomas Popela <tpopela@redhat.com>
Date: Wed, 30 Nov 2016 15:10:32 +0100
Subject: [PATCH 510/595] EEditorWebExtension: Use-after-free when restoring
 the inline images

Restore the inline images when the GVariant that's holding them is valid
(before we pass it to g_dbus_method_invocation_return_value).
---
 modules/webkit-editor/web-extension/e-editor-web-extension.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/modules/webkit-editor/web-extension/e-editor-web-extension.c b/modules/webkit-editor/web-extension/e-editor-web-extension.c
index 85893a9f92..cc37069ec9 100644
--- a/modules/webkit-editor/web-extension/e-editor-web-extension.c
+++ b/modules/webkit-editor/web-extension/e-editor-web-extension.c
@@ -1796,6 +1796,9 @@ handle_method_call (GDBusConnection *connection,
 			g_warning ("Unsupported flags combination (%d) in (%s)", flags, G_STRFUNC);
 		}
 
+		if ((flags & E_CONTENT_EDITOR_GET_INLINE_IMAGES) && from_domain && *from_domain && inline_images)
+			e_editor_dom_restore_images (editor_page, inline_images);
+
 		/* If no inline images are requested we still have to return
 		 * something even it won't be used at all. */
 		g_dbus_method_invocation_return_value (
@@ -1806,9 +1809,6 @@ handle_method_call (GDBusConnection *connection,
 				inline_images ? inline_images : g_variant_new_int32 (0)));
 
 		g_free (value);
-
-		if ((flags & E_CONTENT_EDITOR_GET_INLINE_IMAGES) && from_domain && *from_domain && inline_images)
-			e_editor_dom_restore_images (editor_page, inline_images);
 	} else if (g_strcmp0 (method_name, "DOMInsertHTML") == 0) {
 		const gchar *html;
 
-- 
2.15.0

